app/utils/guaranteeParsing/DefaultGuaranteeApplier.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.guaranteeParsing
+
+import models.ParseError.GuaranteeAmountZero
+import models.{Guarantee, ParseError, SpecialMentionGuarantee, SpecialMentionGuaranteeDetails}
+
+class DefaultGuaranteeApplier {
+
+  def applyDefaultGuarantee(g: Guarantee, s: SpecialMentionGuarantee): Either[ParseError, SpecialMentionGuaranteeDetails] =
+    s.toDetails(g.gReference) match {
+      case Left(error) => Left(error)
+      case Right(details) => details.guaranteeAmount match {
+        case None => Right(details.copy(Some(BigDecimal(10000.00)), Some("EUR")))
+        case Some(amount) => amount match {
+          case a if a.equals(BigDecimal(0)) => Left(GuaranteeAmountZero("GuaranteeAmount cannot be zero"))
+          case _ => Right(details)
+        }
+      }
+    }
```

Comment:
You could simplify this with a `map` on line 25`s.toDetails(g.gReference) `

Reply:
I tried this, but I was having issues with resolving the left further down. Could you show me how to resolve that problem?